### PR TITLE
Refactor validate_vm into 3 different parts and propagate to pre-pro …

### PR DESCRIPTION
**Pull request by jpa198**
…and workflow

Still needs kisti/tacc script updates

#### Checklist
- Have you updated the README (yes/no)?
- Have you updated the CHANGELOG (yes/no)?

### CODE REVIEW

File: scripts/cybershake/install_cybershake_fault.py [hunk 1]
```diff
@@ -159,7 +159,12 @@ def install_fault(
...
-        valid_vm, message = validate_vm.validate_vm(vel_mod_dir, srf=list_srf[0])
+        valid_vm, message = validate_vm.validate_vm_params(vm_params_path)
+        if not valid_vm:
+            message = "Error: VM {} failed {}".format(fault_name, message)
+            logger.log(NOPRINTCRITICAL, message)
+            raise RuntimeError(message)
+        valid_vm, message = validate_vm.validate_vm_files(vel_mod_dir, srf=list_srf[0])
         if not valid_vm:
             message = "Error: VM {} failed {}".format(fault_name, message)
             logger.log(NOPRINTCRITICAL, message)
```
Comment (jasonmotha) [inline scripts/cybershake/install_cybershake_fault.py:170]: Are these lines duplicated?

File: scripts/nesi_scripts/vm_params_gen.sl [hunk 2]
```diff
@@ -54,7 +54,8 @@ end_time=`date +$runtime_fmt`
 echo $end_time
 
 timestamp=`date +%Y%m%d_%H%M%S`
-if [[ -f $OUT_DIR/vm_params.yaml ]]; then
+res=`python $gmsim/qcore/qcore/validate_vm.py params $OUT_DIR/vm_params.yaml`
+if [[ $? == 0 ]]; then
```
Comment (jasonmotha) [inline scripts/nesi_scripts/vm_params_gen.sl:58]: I personally like saving $? the line below the call, just so someone doesn't put a line before the if statement.

Up to you. Having no space between the if and the `res` line might also be enough.

**Review by jasonmotha — CHANGES_REQUESTED** [review]

**Comment by jasonmotha** [comment]
Oh, was this not the one I was meant to review lol?

**Comment by jpa198** [comment]
> Oh, was this not the one I was meant to review lol?

Ugh, I didn't hit the go button to githubs satisfaction. Ready for review though

File: scripts/cybershake/install_cybershake_fault.py (hunk 1 above)
Comment (jpa198) [reply scripts/cybershake/install_cybershake_fault.py:170]: No, one check for params, the other for the files. As that functionality has now been split up, this is maintaining the old post VM install style functionality

**Review by sungeunbae — COMMENTED** [review]
files, params, file : I am happy - just awaiting on the discussion regarding "file" on the qcore PR.

File: scripts/cybershake/install_cybershake_fault.py [hunk 3]
```diff
@@ -159,7 +159,12 @@ def install_fault(
     vel_mod_dir = simulation_structure.get_fault_VM_dir(root_folder, fault_name)
     vm_params_path = os.path.join(vel_mod_dir, VM_PARAMS_FILE_NAME)
     if check_vm:
-        valid_vm, message = validate_vm.validate_vm(vel_mod_dir, srf=list_srf[0])
+        valid_vm, message = validate_vm.validate_vm_params(vm_params_path)
```
Comment (jasonmotha) [inline scripts/cybershake/install_cybershake_fault.py:162]: Ahhh, they are different. Could we use different variable names for clarity of reading 

valid_vm_params, valid_vm_files etc?

File: scripts/nesi_scripts/vm_gen.sl [hunk 4]
```diff
@@ -51,7 +51,7 @@ echo $end_time
 
 timestamp=`date +%Y%m%d_%H%M%S`
 #test before update
-res=`python $gmsim/qcore/qcore/validate_vm.py $OUT_DIR --srf $SRF`
+res=`python $gmsim/qcore/qcore/validate_vm.py files $OUT_DIR`
```
Comment (jasonmotha) [inline scripts/nesi_scripts/vm_gen.sl:54]: Would be good to be consistent with this slurm too with results.

**Review by jasonmotha — CHANGES_REQUESTED** [review]

**Review by jasonmotha — APPROVED** [review]
Sweet!

**Comment by jpa198** [comment]
@sungeunbae

**Review by sungeunbae — APPROVED** [review]

**Pull request merged**